src/main/java/com/minecolonies/coremod/colony/buildings/modules/settings/StringSetting.java

```diff
@@ -107,6 +108,16 @@ public void trigger()
         }
     }
 
+    @Override
+    public void updateSetting(final ISetting iSetting)
```

Comment:
Maybe instead of hard clearing the old values we could check the different items between the 2 lists?
So if the old contains "a", "b", "c" and the new settings would contain "a", "b", "c", "d" it would only append "d" on the end, this way it wouldn't matter if positions of settings swap around because they're always appended instead of potentially inserted somewhere in the center.

Reply:
I thought about that, but if we remove a settingvalue from a building, we actually want it gone from all.